processing/src/main/java/org/apache/druid/segment/data/FixedIndexed.java

```diff
@@ -43,12 +44,14 @@
  * If {@link #hasNull} is set, id 0 is ALWAYS null, so the comparator should be 'nulls first' or else behavior will
  * be unexpected. {@link #hasNull} can only be set if also {@link #isSorted} is set, since the null value is not
  * actually stored in the values section.
+ *
+ * This class is not thread-safe.
```

Comment:
ah i should adjust this to say this class is thread-safe if `read(ByteBuffer buffer, int offset)` is thread-safe